src/components/views/rooms/RoomList.js

```diff
@@ -94,10 +96,12 @@ module.exports = React.createClass({
                 }
                 this._groupStores[tag] = GroupStoreCache.getGroupStore(tag);
                 this._groupStores[tag].registerListener(() => {
```

Comment:
separate problem I know, but does this not leak?

Reply:
oh. Yes, but only when RoomList mounts multiple times.

Reply:
how so? this code is executed every time the `FilterStore` is updated.

Reply:
The condition on `this._groupStores[tag]` prevents more than one listener per group store.

Reply:
:+1: 

as discussed we do end up leaking groupstore listeners at logout, which we should probably fix, but in a separate PR.